File: js/complete/no010.js
#!/usr/bin/env node

/* Find the sum of all the primes below two million. */

var fns = require('../functions.js'),
    operator = require('../operator.js'),
    timer = require('../timer.js');

exports.main = function() {
    var primes = fns.sieve(2000000),
        result = operator.sum(primes);
    return result;
};

if (require.main === module) {
    timer.timer(10, exports.main);
}

File: python_code/complete/no010.py
#!/usr/bin/env python

# Find the sum of all the primes below two million.

from python_code.decorators import euler_timer
from python_code.functions import sieve

def main(verbose=False):
    return sum(sieve(2000000 - 1))

if __name__ == '__main__':
    print euler_timer(10)(main)(verbose=True)
